extensions/shared/plan-upgrade-notification.js

```diff
@@ -15,9 +16,33 @@ if ( undefined !== typeof window && window.location ) {
 	const { query } = parseUrl( window.location.href, true );
 
 	if ( query.plan_upgraded ) {
-		dispatch( 'core/notices' ).createNotice(
-			'success',
-			__( 'Plan upgraded. You may use premium blocks now.', 'jetpack' )
-		);
+		const path = '/jetpack/v4/site';
```

Comment:
This API works well for Jetpack sites but for .com simple sites we seem to need something else as it doesn't exist.

This looks like equivalent and also has `plan.product_name`:

```
https://public-api.wordpress.com/rest/v1.2/sites/SITE_ID
```

<img width="325" alt="image" src="https://user-images.githubusercontent.com/87168/65489917-d4f7cd00-deb4-11e9-86bc-298c55f3b5f1.png">

There's a way to detect if site is wpcom site but I can't find it now — @ockham you prolly know?

Reply:
We've got this at the moment:

https://github.com/Automattic/jetpack/blob/a7e494e0e167507108313b581764d34a68cca6d3/extensions/shared/components/upgrade-nudge/index.jsx#L65

Whether we should encourage reuse of that pattern is something I'm undecided on 😬  

Revisiting the way we pass data via `Jetpack_Some_Global_Variable` to our blocks is something that would be good to do, we should consider using a wp-data store. The environment (wpcom/jetpack…) is something that makes sense as part of that store and not another global variable. Alas, this change probably isn't enough to merit reworking that bit 🤷‍♂

Reply:
Added a helper as a first step something easier: https://github.com/Automattic/jetpack/pull/13525

Turns out I need something like that for Tiled gallery's description as well: D32972-code